app/utils/transactionHelpers.ts

```diff
@@ -571,8 +572,12 @@ export function buildTransactionAccountSignature(
     return transactionAccountSignature;
 }
 
-export function isSuccessfulTransaction(event: TransactionSummary) {
-    return event.result.outcome === 'success';
+export function isSuccessfulTransaction(summary: BlockItemSummary) {
+    if (summary.type === TransactionSummaryType.AccountTransaction) {
+        return summary.transactionType !== TransactionKindString.Failed;
+    }
+    // other block items don't fail // TODO is this completely true
```

Comment:
We need to resolve this TODO.

Reply:
Yes, it is true.
The only reason to have failed account transactions is to make the account pay, but that is not relevant for other types, so they wouldn't be put in chain.